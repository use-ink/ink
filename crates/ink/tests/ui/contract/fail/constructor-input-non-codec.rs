#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[::ink::scale_derive(TypeInfo)]
    pub struct NonCodecType;

    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor(_input: NonCodecType) -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
