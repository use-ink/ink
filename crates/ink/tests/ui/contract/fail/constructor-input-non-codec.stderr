error[E0277]: the trait bound `NonCodecType: WrapperTypeDecode` is not satisfied
  --> tests/ui/contract/fail/constructor-input-non-codec.rs:11:28
   |
11 |         pub fn constructor(_input: NonCodecType) -> Self {
   |                            ^^^^^^^^^^^^^^^^^^^^ the trait `WrapperTypeDecode` is not implemented for `NonCodecType`, which is required by `NonCodecType: ink::parity_scale_codec::Decode`
   |
   = help: the following other types implement trait `WrapperTypeDecode`:
             Arc<T>
             Box<T>
             Rc<T>
   = note: required for `NonCodecType` to implement `ink::parity_scale_codec::Decode`
note: required by a bound in `DispatchInput`
  --> src/codegen/dispatch/type_check.rs
   |
   | pub struct DispatchInput<T>(T)
   |            ------------- required by a bound in this struct
   | where
   |     T: scale::Decode + 'static;
   |        ^^^^^^^^^^^^^ required by this bound in `DispatchInput`

error[E0277]: the trait bound `NonCodecType: ink::parity_scale_codec::Decode` is not satisfied
  --> tests/ui/contract/fail/constructor-input-non-codec.rs:11:9
   |
11 | /         pub fn constructor(_input: NonCodecType) -> Self {
12 | |             Self {}
13 | |         }
   | |_________^ the trait `WrapperTypeDecode` is not implemented for `NonCodecType`, which is required by `NonCodecType: ink::parity_scale_codec::Decode`
   |
   = help: the following other types implement trait `WrapperTypeDecode`:
             Arc<T>
             Box<T>
             Rc<T>
   = note: required for `NonCodecType` to implement `ink::parity_scale_codec::Decode`

error[E0277]: the trait bound `NonCodecType: Encode` is not satisfied
  --> tests/ui/contract/fail/constructor-input-non-codec.rs:1:1
   |
1  |   #[ink::contract]
   |   ^^^^^^^^^^^^^^^^ the trait `WrapperTypeEncode` is not implemented for `NonCodecType`, which is required by `NonCodecType: Encode`
...
11 | /         pub fn constructor(_input: NonCodecType) -> Self {
12 | |             Self {}
13 | |         }
   | |_________- required by a bound introduced by this call
   |
   = help: the following other types implement trait `WrapperTypeEncode`:
             &T
             &mut T
             Arc<T>
             Box<T>
             Cow<'a, T>
             Rc<T>
             String
             Vec<T>
             ink::parity_scale_codec::Ref<'a, T, U>
   = note: required for `NonCodecType` to implement `Encode`
note: required by a bound in `ExecutionInput::<ArgumentList<ArgumentListEnd, ArgumentListEnd>>::push_arg`
  --> $WORKSPACE/crates/env/src/call/execution.rs
   |
   |     pub fn push_arg<T>(
   |            -------- required by a bound in this associated function
...
   |         T: scale::Encode,
   |            ^^^^^^^^^^^^^ required by this bound in `ExecutionInput::<ArgumentList<ArgumentListEnd, ArgumentListEnd>>::push_arg`
